* Rmail can now handle capitilized month names.
  Patch: fix-rmail-capitalized-month-names.diff
  Provided-by: Sven Joachim <svenjoac@gmx.de>
  Originally-reported-by: Jeroen Nijhof <jeroen@nijhof.uklinux.net>
  Date: Thu, 06 Aug 2009 22:51:47 +0200
  Added-by: Rob Browning <rlb@defaultvalue.org>
  Status: incorporated upstream
  Bug: 540234

  Rmail can not handle Date lines that have a capitalized month name
  like "Date: Thu, 06 Aug 2009 18:33:32 +0100".

Index: b/lisp/mail/rmailsum.el
===================================================================
--- a/lisp/mail/rmailsum.el
+++ b/lisp/mail/rmailsum.el
@@ -518,16 +518,19 @@
    (concat (save-excursion
 	     (if (not (re-search-forward "^Date:" nil t))
 		 "      "
-	       (cond ((re-search-forward "\\([^0-9:]\\)\\([0-3]?[0-9]\\)\\([- \t_]+\\)\\([adfjmnos][aceopu][bcglnprtvy]\\)"
-		       (line-end-position) t)
+	       ;; Match month names case-insensitively
+	       (cond ((let ((case-fold-search t))
+			(re-search-forward "\\([^0-9:]\\)\\([0-3]?[0-9]\\)\\([- \t_]+\\)\\([adfjmnos][aceopu][bcglnprtvy]\\)"
+					   (line-end-position) t))
 		      (format "%2d-%3s"
 			      (string-to-number (buffer-substring
 						 (match-beginning 2)
 						 (match-end 2)))
 			      (buffer-substring
 			       (match-beginning 4) (match-end 4))))
-		     ((re-search-forward "\\([^a-z]\\)\\([adfjmnos][acepou][bcglnprtvy]\\)\\([-a-z \t_]*\\)\\([0-9][0-9]?\\)"
-		       (line-end-position) t)
+		     ((let ((case-fold-search t))
+			(re-search-forward "\\([^a-z]\\)\\([adfjmnos][acepou][bcglnprtvy]\\)\\([-a-z \t_]*\\)\\([0-9][0-9]?\\)"
+					   (line-end-position) t))
 		      (format "%2d-%3s"
 			      (string-to-number (buffer-substring
 						 (match-beginning 4)
